### DISCUSSION TITLE
Wrong URL for GitHub pages

### DISCUSSION THREAD

**Discussion by ghost** [community/community#37617] [Other Feature Feedback, Questions, & Ideas]
If anyone could help I'd be really grateful as I've searched all of the forums and cannot find anyone else with the same problem.

Basically, I went to deploy another webpage on GitHub pages and it returned 404. I thought it was because I was using React and did the necessary steps to make it work, but still 404. I then checked my other GItHub pages and they all return 404.

So, I checked the URL's and they seem very wrong.  It says: Your site is live at http:// {followed  by my portfolio page} / {followed by the repository name}.

I cannot understand what has gone wrong here. I created the portfolio about a month ago, pushing it to GiHub and then used the free .me domain name as part of GitHub student on NameCheap.

Please Help!
thanks for your time,
Mark

**Comment by DSDmark** [comment]
Hey @houlli82,

I'm not sure, but you could try these tips:-

First open your repo and click on settings, after that there is a **Code and automation** section. In that section there in a **pages** click on that, and it should be looked like that ok. if you are using **gh-pages**.

![Screenshot from 2022-10-28 15-37-18](https://user-images.githubusercontent.com/85744816/198562740-357dbc30-5432-428f-bf82-2b85be6a2159.png)

**Reply by DSDmark** [reply]
And also if you can add your repo link then it would be great for understanding your problem

**Reply by ghost** [reply]
Hi @DSDmark,

Thank you  for your reply.

I had a look under Branch and in the dropdown there is only 'master' or 'main', but there isn't gh-pages.

**Reply by ghost** [reply]
This is the repo I'm trying to push to pages although all of my repo's have this same problem.
https://github.com/houlli82/Monster-Rolodex

Thanks houlli82

**Reply by ghost** [reply]
<img width="948" alt="github-problem" src="https://user-images.githubusercontent.com/92862939/198564352-b57cbb0e-2359-4eea-b87f-52a67c016cc8.png">

**Reply by DSDmark** [reply]
Hey @houlli82,

Do you know when we deploy ReactJS app, we need to build project first?.
There is no build folder.

check this out :- https://github.com/gitname/react-gh-pages

**Reply by ghost** [reply]
I see, I will read the documentation now, thank you!

I have the same problem on my other repositories too that weren't built with React.  Where it says "Your site is live at..." it has my portfolio.me page and not user-name.github.io/

Thanks again.

**Comment by jfullstackdev** [comment]
this is the link you presented 

![image](https://user-images.githubusercontent.com/47092464/198570286-04c5198d-d43c-4be3-af06-7163166fa899.png)

and also this is the root of your domain, I'm just wondering if you use React in deploying it, if you both use React then at least you have an idea what's going wrong

![image](https://user-images.githubusercontent.com/47092464/198570437-a109140f-c5b6-440f-ab77-5bf6dd447aaa.png)

**Reply by ghost** [reply]
Thanks for taking a look @jdevstatic,

I didn't use React to deploy my .me site.  

Cheers

**Comment by airtower-luna** [comment]
The reason for the different domain is pretty simple: Your portfolio repository is called `houlli82.github.io`. The `<username>.github.io` pattern makes it your Pages *user site*, and any other Pages side you add will be a subdirectory of that site, *unless* you give it its own domain.

Just rename the portfolio repository so it won't be your user site any more. :wink:

**Reply by ghost** [reply]
Hi @airtower-luna,

Thanks for taking a look.  

I see, I don't know how I missed that. I did as you said and changed the repository name, but the problem still persists.  There is my .me at the start of all of my pages URL when I launch the site from GitHub, however the path looks correct under settings/pages.

I tried copying the URL into my browser, but it redirects to the .me site still.

Thanks again!

**Reply by airtower-luna** [reply]
Maybe try triggering another deployment on the repository that has the `.me` domain, to make sure the deployment is up to date.

If that (and a fresh deployment of your new repo after) doesn't help you might want to [contact GitHub support](https://support.github.com/request), just in case something is stuck behind the scenes.

**Reply by ghost** [reply]
Thanks @airtower-luna!

I'm still not sure how to fix this, so I've opened a ticket with GitHub, hopefully someone will respond soon.

I'll report back here if/when it's fixed.

Thanks again.
